### ISSUE TITLE
[Desktop] Crypto Wallet account names change after reset

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#11705]
## Description 

Changing built-in Crypto wallet account names in Brave does not stick after the browser is closed/restarted. After restart, all but the first account names get reset to their default titles. 

## Steps to Reproduce

As per original user report:

1. Open “Crypto Wallet” (built-in Metamask)
2. Add more than 1 accounts to wallet
3. Rename all accounts
4. Restart Brave
5. Only the first account name will stick, the other account names will be changed to default (e.g. Account2, Account3)

## Actual result:

Wallet/account names are reset to their default names (except for the first one).

## Expected result:
 All account names should remain after restart

## Reproduces how often: 

Easily for the user. I was unable to reproduce the issue.

## Brave version (brave://version info)

Note: The user reported in the thread that they were running version `Version 1.12.114 Chromium: 84.0.4147.135 (Official Build) (64-bit)`. I asked them to update and see if the problem persisted but they have not yet responded. 

## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel? N/A
- Can you reproduce this issue with the nightly channel? N/A

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? N/A
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

## Miscellaneous Information:

Original user thread: 
https://community.brave.com/t/wallet-account-names-change-after-restarting-brave/156883

**Comment by FSM1** [comment]
I am experiencing the exact same on Win 10 Version 1.14.84

**Comment by O1O1O1O** [comment]
Still an issue in 1.18.62 - very annoying...

**Comment by autsauce** [comment]
Issue still persisting.

Windows 10
Version 1.21.76 Chromium: 89.0.4389.86 (Official Build) (64-bit)

**Comment by srirambv** [comment]
I am unable to repro  the issue with steps from issue description. Can someone give some consistent steps to repro the issue I can give it check on my end

**Comment by autsauce** [comment]
@srirambv when I follow those exact same steps, upon reopening the browser, all but 1 of my accounts are reset to "Account 2", "Account 3", etc. The only apparent difference is that the one account that keeps its name happens to be an imported wallet. I'm not sure if that is relevant.

**Comment by autsauce** [comment]
@srirambv any update on this? I know it doesn't seem like a major issue, but I have multiple wallets for personal/business use and it's making it difficult to keep them distinguished. Could the trouble replicating the issue be due to different browser settings or discrepancies in OS/browser versions?

**Comment by srirambv** [comment]
> Could the trouble replicating the issue be due to different browser settings or discrepancies in OS/browser versions?

Ideally it shouldn't. @autsauce can you please add your browser version and the platform you are using? 

If you can go through the video i posted above and confirm this is exactly what you have done or if you can give the OS/Component details and then give exact steps that you performed so I can give it a try on my end to see what's going on

**Comment by autsauce** [comment]
@srirambv I confirm that I am performing the exact same steps as you depicted in the video. Except when I restart the browser, all but one of my wallet names are reset. I can take a video if you would like. My browser/OS details are below, let me know if you require some other more specific information.

Brave: Version 1.21.77 Chromium: 89.0.4389.90 (Official Build) (64-bit)
OS: Microsoft Windows 10 Pro

**Comment by srirambv** [comment]
@autsauce  if you can record the steps on your machine that would be very helpful to see whats exactly going on. I'll try to repro the issue on Windows as well and see if I have any luck

**Comment by autsauce** [comment]
@srirambv 

https://user-images.githubusercontent.com/46509848/111823862-d5133600-88bb-11eb-91a7-6fb566c61165.mov

**Comment by srirambv** [comment]
@autsauce I see the issue. I wonder if the issue is happening beacuse your primary account is imported (via Private key?) and later you have added the other two devices. Let me give this scenario a try and see if it reproduces

**Comment by autsauce** [comment]
@srirambv yes one of the wallets was imported via private key. It's actually the last wallet I added to Brave Wallets, however.

**Comment by srirambv** [comment]
>It's actually the last wallet I added to Brave Wallets, however.

Its weird that its showing up as Account 1 if you have added it as the last one. I can't get to repro imported wallet to be the first wallet. I need some time to try this out again. Will keep the thread posted on my findings

**Comment by srirambv** [comment]
I was able to repro the issue but not very sure of why/how it happens. So steps seems to be like this 

1. Create a new wallet
2. Change the name to `Personal` or whatever you want
3. Import a new wallet by entering the private key
4. **If you get an error after entering the imported key then the issue repro's. Note: Even though there was an error while importing wallet was still imported**
5. Rename the imported wallet as `Imported`
6. Personal wallet name is still retained
7. Lock wallet and unlock, Personal account name is changed to `Account 2` and Imported wallet retains the name

Tried adding the name via contacts and was able to change it but it still reverted back to `Account 2`
![image](https://user-images.githubusercontent.com/17010094/112144466-2af42080-8bd1-11eb-8e6c-3af9887a89df.png)


On selecting General throws an error message like this 
![image](https://user-images.githubusercontent.com/17010094/112144027-91c50a00-8bd0-11eb-8259-8fb034ae1e3b.png)

The error message has the wallet  address from step 1/2(Personal account)

cc: @bbondy @Douglashdaniel 

@autsauce can you check if you see the same issue (error message) when you try to change the account name and go to general settings

**Comment by srirambv** [comment]
Console logs the following error as well 
```
Uncaught (in promise) TypeError: Cannot use 'in' operator to search for '<imported wallet address>' in false
    at r.default.setSelectedAddress (background.js:1)
    at r.default.syncAddresses (background.js:1)
    at le._onKeyringControllerUpdate (background.js:1)
    at t.exports.<anonymous> (background.js:1)
    at c (background.js:111)
    at t.exports.s.emit (background.js:111)
    at t.exports.putState (background.js:109)
    at t.exports.updateState (background.js:109)
    at E._updateMemStoreKeyrings (background.js:1)
    at async Promise.all (index 3)
    at async E.unlockKeyrings (background.js:1)
    at async le.submitPassword (background.js:1)
```

**Comment by srirambv** [comment]
Same error shows up for each account when trying to edit the name 

![image](https://user-images.githubusercontent.com/17010094/112149860-5974fa00-8bd7-11eb-89d9-5688b56707e9.png) | ![image](https://user-images.githubusercontent.com/17010094/112149699-2a5e8880-8bd7-11eb-9a9e-8f7d20697829.png)
-- | --

**Comment by autsauce** [comment]
@srirambv I'm glad you were able to repro the issue. I tried changing my account name and then going to general settings, but I did not see the error message that you received.

**Comment by Ruaphoc** [comment]
+1
Brave Version 1.29.79 Chromium: 93.0.4577.63 (Official Build) (x86_64)
System Version:	macOS 11.5.2 (20G95)

**Comment by srirambv** [comment]
+1 from @DefiDebauchery  via #18026

**Comment by Douglashdaniel** [comment]
Closing since our Crypto Wallet `Ethereum Remote Client Extension` is no longer maintained.
